### DISCUSSION TITLE
Creating cylinders given start and end position

### DISCUSSION THREAD

**Discussion by TheRiverBank** [IfcOpenShell/IfcOpenShell#2803] [Q&A]
Not sure if this is the right place to ask, but I am trying to create cylinders given some start and end pos in 3D and apply color.
I have tried using the same setup given here [https://github.com/IfcOpenShell/files/blob/master/cylinders.ifc](https://github.com/IfcOpenShell/files/blob/master/cylinders.ifc) but it seems like these are created based on a given rotation around a point. I did not manage to alter that data to work with start and end positions. As I'm completely new to creating IFC files my question is how can I make this data produce cylinders given a start and end pos? I also found the [IfcRightcircularCylinder](https://standards.buildingsmart.org/IFC/RELEASE/IFC2x3/TC1/HTML/ifcgeometricmodelresource/lexical/ifcrightcircularcylinder.htm) which I thought could be useful but I have not managed to create a working solution with this one either.

**Comment by Moult** [comment]
Have you considered using the IfcOpenShell API? There are a number of geometry functions available to do what you're after.
